### DISCUSSION TITLE
GH CLI release asset binary upload error

### DISCUSSION THREAD

**Discussion by eighty4** [community/community#54896] [Programming Help]
### Body

I think there's an error in the [docs on uploading release assets with the GH CLI](https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#upload-a-release-asset), which prescribe this command:

```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  --hostname HOSTNAME \
  /repos/OWNER/REPO/releases/RELEASE_ID/assets?name=example.zip \
  -f '@example.zip'
```

Using `-f @file` in this way throws a very nondescript `invalid key` error; the `gh cli -h` docs state -f is for key value pairs, yet the docs do not show a key with the `@file` arg value.

I didn't want to spend more time diving into github.com/cli/cli source to see if there was a magic -f key for posting binary as the request body, so I switched to the cURL method. I'm not sure if this is a docs issue or a CLI bug, so I'm reporting it here instead of github.com/cli/cli/issues.

Thanks!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by marianhlavac** [comment]
Same here, can this be clarified?

It's also worth noting that `gh release upload` can be used instead: https://cli.github.com/manual/gh_release_upload

**Comment by teebszet** [comment]
more info.. if I try with:

```
--hostname "uploads.github.com" \
"/repos/{owner}/{repo}/releases/RELEASE_ID/assets?name=example.zip" \
-F "body=@example.zip"
```

I get a tls error with the CLI trying to upload to api.uploads.github.com

```
Post "https://api.uploads.github.com/.../assets?name=example.zip": tls: failed to verify certificate: x509: certificate is valid for *.github.io, github.io, *.github.com, github.com, www.github.com, *.githubusercontent.com, githubusercontent.com, not api.uploads.github.com
```

**Reply by fieldsfarmer** [reply]
hostname should github.com
